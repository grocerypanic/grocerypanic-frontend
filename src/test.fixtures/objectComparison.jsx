export const propCount = (hash, length) => {
  expect(Object.keys(hash).length).toBe(length);
};
