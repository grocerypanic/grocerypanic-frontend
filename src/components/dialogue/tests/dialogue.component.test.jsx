import React from "react";
import { render, cleanup, waitFor, fireEvent } from "@testing-library/react";
import { propCount } from "../../../test.fixtures/objectComparison";

import Header from "../../header/header.component";
import Dialogue from "../dialogue.component";

import calculateMaxHeight from "../../../util/height";

jest.mock("../../header/header.component");
jest.mock("../../../util/height");

Header.mockImplementation(() => <div>MockHeader</div>);
calculateMaxHeight.mockImplementation(() => 200);
const Footer = jest.fn(() => <div>MockFooter</div>);

const defaultProps = {
  title: "mockTitle",
  headerTitle: "mockHeaderTitle",
  body: "mockbody1\nmockbody2",
  Footer: Footer,
};

describe("Setup Environment", () => {
  let current;
  let utils;

  beforeEach(() => {
    jest.clearAllMocks();
    current = {
      ...defaultProps,
    };
  });

  afterEach(cleanup);

  const renderHelper = (currentProps) => {
    return render(<Dialogue {...currentProps} />);
  };

  describe("when passed parameters", () => {
    beforeEach(() => {
      utils = renderHelper(current);
    });

    it("renders, should call header with the correct params", () => {
      expect(Header).toHaveBeenCalledTimes(1);

      const headerCall = Header.mock.calls[0][0];
      propCount(headerCall, 2);
      expect(headerCall.title).toBe(defaultProps.headerTitle);
      expect(headerCall.transaction).toBe(false);
    });

    it("renders, should display the title", () => {
      expect(utils.getByText(defaultProps.title)).toBeTruthy();
    });

    it("renders, should display the body", () => {
      defaultProps.body.split("\n").forEach((o) => {
        expect(utils.getByText(o)).toBeTruthy();
      });
    });

    it("renders, should display the footer object", () => {
      expect(Footer).toBeCalledTimes(1);
    });

    it("renders, should call calculateMaxHeight", () => {
      expect(calculateMaxHeight).toBeCalledTimes(1);
    });

    it("renders, should call calculateMaxHeight on a resize event", async (done) => {
      expect(calculateMaxHeight).toBeCalledTimes(1);
      fireEvent(window, new Event("resize"));
      await waitFor(() => expect(calculateMaxHeight).toBeCalledTimes(2));
      done();
    });

    it("should match the snapshot on file (styles)", () => {
      expect(utils.container).toMatchSnapshot();
    });
  });
});
