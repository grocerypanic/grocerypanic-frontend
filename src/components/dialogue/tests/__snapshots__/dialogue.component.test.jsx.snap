// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment when passed parameters should match the snapshot on file (styles) 1`] = `
.c0 {
  margin-top: calc(50px + 26px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style-type: none;
  list-style-position: outside;
  overflow: scroll;
  max-height: 200px;
  border-style: solid;
  border-color: #808080;
  border-width: 10px;
}

.c4 {
  padding-left: 0px;
  max-width: 250px;
  background: #808080;
  padding-top: 0px;
}

.c2 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: #000000;
  font-size: 15px;
  min-width: 220px;
}

.c5 span {
  margin-bottom: 10px;
}

.c5 span:last-child {
  margin-bottom: 0px;
}

<div>
  <div>
    MockHeader
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 alert alert-success"
      >
        <div>
          mockTitle
        </div>
      </div>
      <div
        class="c3 overflow-auto"
        size="200"
      >
        <div
          class="c4"
        >
          <li
            class="c5"
          >
            <span>
              mockbody1
            </span>
            <span>
              mockbody2
            </span>
          </li>
        </div>
      </div>
    </div>
    <div>
      MockFooter
    </div>
  </div>
</div>
`;
